### PR TITLE
Delete consumers after test run

### PR DESCRIPTION
**Pull request by stanlyDoge**
Fixes https://github.com/strimzi/strimzi-kafka-bridge/issues/335

### CODE REVIEW

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerGeneratedNameTest.java [hunk 1]
```diff
@@ -50,18 +58,23 @@ void createConsumerNameNotSet(VertxTestContext context) {
...
                         JsonObject bridgeResponse = response.body();
                         String consumerInstanceId = bridgeResponse.getString("instance_id");
+                        name.set(consumerInstanceId);
                         assertTrue(consumerInstanceId.startsWith("kafka-bridge-consumer-"));
+                        create.complete(true);
                     });
-                    context.completeNow();
                 });
+        create.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+        consumerService().deleteConsumer(context, groupId, name.get());
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerGeneratedNameTest.java]: let's use the same format as for other tests, putting the method call on the next line.

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerGeneratedNameTest.java [hunk 2]
```diff
@@ -37,9 +43,11 @@ static void revertUnsetBridgeID() {
     }
 
     @Test
-    void createConsumerNameNotSet(VertxTestContext context) {
+    void createConsumerNameNotSet(VertxTestContext context) throws InterruptedException, ExecutionException, TimeoutException {
         JsonObject json = new JsonObject();
+        AtomicReference<String> name = new AtomicReference<>();
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerGeneratedNameTest.java]: can't we just use a field member as we do for `ConsumerTest` class?

```java
private String name:
```

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerGeneratedNameTest.java [hunk 3]
```diff
@@ -73,8 +86,11 @@ void createConsumerNameSet(VertxTestContext context) {
                         JsonObject bridgeResponse = response.body();
                         String consumerInstanceId = bridgeResponse.getString("instance_id");
                         assertTrue(consumerInstanceId.equals("consumer-1"));
+                        create.complete(true);
                     });
-                    context.completeNow();
                 });
+        create.get(TEST_TIMEOUT, TimeUnit.SECONDS);
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerGeneratedNameTest.java]: `deleteConsumer` on the next line

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 4]
```diff
@@ -51,10 +53,11 @@
     @Test
     void createConsumer(VertxTestContext context) throws InterruptedException, TimeoutException, ExecutionException {
         // create consumer
-        consumerService().createConsumer(context, groupId, consumerWithEarliestReset);
+        ConsumerService consumer = consumerService().createConsumer(context, groupId, consumerWithEarliestReset);
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: the `consumerService()` return a singleton, it's useless to save it in a variable.

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 5]
```diff
@@ -51,10 +53,11 @@
...
 
         context.completeNow();
         assertTrue(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS));
+        consumer.deleteConsumer(context, groupId, name);
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: just call `consumerService().deleteConsumer(....)` but with `deleteConsumer` on the next line :-)

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 6]
```diff
@@ -208,11 +211,12 @@ void createConsumerWithForwardedHeaders(VertxTestContext context) throws Interru
                     context.completeNow();
                 });
 
+        consumerService().deleteConsumer(context, groupId, name);
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: Two things:

* I don't think you can do in this way when `context.completeNow' is called async and could end the test before deleting. You need a `CompletableFuture` as done in other tests.
* delete on the next line :-)

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 7]
```diff
@@ -234,11 +238,12 @@ void createConsumerWithForwardedHeader(VertxTestContext context) throws Interrup
...
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: all the same comments as the method before.

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 8]
```diff
@@ -265,12 +270,13 @@ void createConsumerWithMultipleForwardedHeaders(VertxTestContext context) throws
...
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: all the same comments as the method before.

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 9]
```diff
@@ -294,11 +300,12 @@ void createConsumerWithForwardedPathHeader(VertxTestContext context) throws Inte
...
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: all the same comments as the method before.

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 10]
```diff
@@ -320,11 +327,12 @@ void createConsumerWithForwardedHeaderDefaultPort(VertxTestContext context) thro
...
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: all the same comments as the method before.

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 11]
```diff
@@ -910,6 +918,8 @@ void consumerAlreadyExistsTest(VertxTestContext context) throws InterruptedExcep
                     create3Again.complete(true);
                 });
 
+        consumerService().deleteConsumer(context, groupId, name);
+        consumerService().deleteConsumer(context, groupId, name + "diff");
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: you should first wait for `create3Again` to be set and then you can deleting consumers.
Again, delete method on next line :-)

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java [hunk 12]
```diff
@@ -1301,6 +1317,7 @@ void createConsumerBridgeIdAndNameSpecified(VertxTestContext context) {
                     });
                     context.completeNow();
                 });
+        consumerService().deleteConsumer(context, groupId, "consumer-1");
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java]: You have to wait on a `CompletableFuture` before deleting, see one of the comments already made above.
As usual, delete method on the next line.

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerGeneratedNameTest.java (hunk 2 above)
Comment (stanlyDoge) [reply src/test/java/io/strimzi/kafka/bridge/http/ConsumerGeneratedNameTest.java]: I am not sure what do you mean :( the name is generated, thus it is something like `kafka-bridge-consumer-564ecf654ef54c65a564ef`. We do not know the generated name before the test is executed.

**Comment by ppatierno** [comment]
@stanlyDoge there is a checkstyle issue to fix

**Pull request merged**